lib/otel/segments/server.js

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const {
+  SEMATTRS_HTTP_METHOD,
+  SEMATTRS_HTTP_ROUTE,
+  SEMATTRS_HTTP_URL,
+  SEMATTRS_RPC_SYSTEM,
+  SEMATTRS_RPC_SERVICE,
+  SEMATTRS_RPC_METHOD
+} = require('@opentelemetry/semantic-conventions')
+const { DESTINATIONS } = require('../../config/attribute-filter')
+const DESTINATION = DESTINATIONS.TRANS_COMMON
+const Transaction = require('../../transaction')
+const urltils = require('../../util/urltils')
+const url = require('url')
+
+module.exports = class ServerSegment {
+  constructor(agent, otelSpan) {
+    this.agent = agent
+    this.transaction = new Transaction(agent)
+    this.transaction.type = 'web'
+    this.otelSpan = otelSpan
+    const rpcSystem = otelSpan.attributes[SEMATTRS_RPC_SYSTEM]
+    const httpMethod = otelSpan.attributes[SEMATTRS_HTTP_METHOD]
+    if (rpcSystem) {
+      this.segment = this.rpcSegment(rpcSystem)
+    } else if (httpMethod) {
+      this.segment = this.httpSegment(httpMethod)
+    } else {
+      this.segment = this.genericHttpSegment()
+    }
+    this.transaction.baseSegment = this.segment
+    return { segment: this.segment, transaction: this.transaction }
```

Comment:
I started writing `class ConsumerSegment` this morning and when I got to this equivalent line, I realized that we are returning something _other_ than the object being created with `new Whatever`. Clearly that doesn't matter since [return overriding](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties#returning_overriding_object) is a thing, but do you think we should do so?

I ended up switching to writing a module `segments/consumer.js` that would just contain all of the code specific to that segment type and exporting a `createConsumerSegment` function.

I'm not opposed to either method. It just feels a little off to me to return something different from a `class` syntax constructor.

Reply:
agreed. let me think on it more.  We can chat tomorrow as well

Reply:
we agreed to address this in a separate PR